src/trunk/apps/fdsnws/fdsnws/http.py

```diff
@@ -51,24 +51,31 @@ def renderErrorPage(request, code, msg, ro=None):
 %s
 """
 
+		noContent = code == http.NO_CONTENT
+
 		# rewrite response code if requested and no data was found
-		if ro is not None and code == http.NO_CONTENT:
+		if noContent and ro is not None:
 			code = ro.noData
 
-		# status code 204 requires no message body
+		# set response code
 		request.setResponseCode(code)
+
+		# status code 204 requires no message body
 		if code == http.NO_CONTENT:
```

Comment:
This would be wrong in case the user requested `nodata=404`. In this case noContent is `true` and code is set to `404` and we want to report the error in a standard FDSNWS error message instead of a 204 with empty body.

Reply:
But a few lines above they are defined to be equivalent?

    noContent = code == http.NO_CONTENT

Reply:
Yes, but L58 in between sets the code to the value requested by the user.

Reply:
Gotcha